File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int left=0;
        int right=0;
        int n=s.length;
        int m=g.length;
        int count=0;
        while(left<m && right<n){
            if(g[left]<=s[right]){
                left++;
                right++;
                count++;

            }
            else{
                right++;
            }
        }
        return count;
    }
}

File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        int left=0;
        int right=0;
        int n=g.size(); int m=s.size();
        int count=0;
        while(left<n && right<m){
            if(s[right]>=g[left]){
                right++;
                left++;
                count++;
            }
            else right++;
        }
        return count;
    }
};